lms/djangoapps/student_enrollment/enrollment.py

```diff
@@ -180,21 +180,46 @@ def enroll(self):
                 )
                 continue
 
+            # check if the enrollment is a change of the originally
+            # enrolled specialisation
+            specialization_change = student["Specialisation_Change_Requested_Within_7_Days"]
+
             # Get the user, the user's password, and their enrollment type
             user, password, enrollment_type = get_or_register_student(
                 student['Email'], student['Email'])
 
             # Get the code of the specialisation to be enrolled in, as well
-            # as the current programme to be susequently unenrolled from
+            # as the current programme to be subsequently unenrolled from
             current_program = student["Programme_ID"]
-            specialisation_to_enroll = student['Specialisation_programme_id']
+            specialization_to_enroll = student['Specialisation_programme_id']
```

Comment:
we sure the field exists for all students?

Reply:
@gomyar Yes.
1. The field exists in CRM for all students.
2. The SPECIALISATION_ENROLLMENT_QUERY (see lms/djangoapps/student_enrollment/zoho.py) only gets student records that have the field populated, so the script never runs on those that don't. 
![image](https://user-images.githubusercontent.com/45522344/162722054-5c4e63a2-c797-498b-8739-1d6d229fb4cc.png)